src/lib.rs

```diff
@@ -215,6 +217,16 @@ fn check_input_on_boot() {
 
 #[skyline::hook(offset = offsets::initial_loading(), inline)]
 fn initial_loading(_ctx: &InlineCtx) {
+    unsafe {
+        match minreq::get("https://coolsonickirby.com/arc/news").send() {
+            Ok(resp) => match resp.json::<HashMap<String, String>>() {
+                Ok(info) => NEWS_DATA.extend(info),
+                Err(err) => println!("{:?}", err),
+            },
+            Err(err) => println!("{:?}", err),
+        }
+    }
+
```

Comment:
Make it a separate method and lock the call behind the online feature mentioned in the other comment (look at check_for_changelog for how to do that)